### PR TITLE
implement primary clipboard pasting using middle-click

### PR DESCRIPTION
**Pull request by jmi2k**
Very simple change to allow middle-click primary clipboard pasting.

I'd like to get some feedback, as adding function calls and `if`s to `button_press_event` seems a bit messy. Also it's not very future-proof, as it keeps calling `send_click`, which could have unexpected side-effects on the long run.

### CODE REVIEW

File: src/EditView.vala [hunk 1]
```diff
@@ -298,6 +301,14 @@ class EditView: Gtk.DrawingArea, Gtk.Scrollable {
 		});
 	}
 
+	public void paste_primary() {
```
Comment (eyelash) [inline src/EditView.vala]: maybe this method should be private?

**Comment by eyelash** [comment]
I guess at some point in the future the whole event handling stuff will be ported to Gtk.Gesture (#34) so the approach here seems to be fine for now.

File: src/EditView.vala (hunk 1 above)
Comment (jmi2k) [reply src/EditView.vala]: It seems so. I'll be honest: as it's my first time with Vala I've tried to make the least changes possible, so I just copy-pasted `paste` and edited the constant.

File: src/EditView.vala [hunk 2]
```diff
@@ -212,6 +212,9 @@ class EditView: Gtk.DrawingArea, Gtk.Scrollable {
 					core_connection.send_gesture(view_id, line, column, "toggle_sel");
 				} else {
 					core_connection.send_click(view_id, line, column, 0, 1);
+					if (event.button == 2) {
```
Comment (eyelash) [inline src/EditView.vala]: Is there no constant in Gtk that can be used instead of the literal `2`?

Comment (jmi2k) [reply src/EditView.vala]: AFAIK there isn't. From the [documentation](https://developer.gnome.org/gdk3/stable/gdk3-Event-Structures.html#GdkEventButton):

> **guint button:** the button which was pressed or released, numbered from 1 to 5. Normally button 1 is the left mouse button, 2 is the middle button, and 3 is the right button. On 2-button mice, the middle button can often be simulated by pressing both mouse buttons together.

Comment (eyelash) [reply src/EditView.vala]: I found it. It's `Gdk.BUTTON_MIDDLE`. Please use that instead of the literal `2`.

**Comment by eyelash** [comment]
Thanks!

**Pull request merged**